h/models.py

```diff
@@ -115,13 +115,25 @@ def __acl__(self):
     }
     __analysis__ = {
         'filter': {
-            'uri': {
+            'uri_index': {
                 'type': 'pattern_capture',
                 'preserve_original': '1',
                 'patterns': [
                     '([^\\/\\?\\#\\.]+)',
                     '([a-zA-Z0-9]+)(?:\\.([a-zA-Z0-9]+))*',
                     '([a-zA-Z0-9-]+)(?:\\.([a-zA-Z0-9-]+))*',
+                    '://(.+)',
+                    '://(.+)\\?',
+                    '://(.+)\\#'
+                ]
+            },
+            'uri_search': {
+                'type': 'pattern_capture',
+                'preserve_original': '1',
+                'patterns': [
+                    '://(.+)',
+                    '://(.+)\\?',
```

Comment:
> Doesn't our various selectors are for making this harder to happen? (I mean we have a selector which saves the text before and after the selected part)

Our selectors are OR-es; not AND-ed. In other words, if any of the matches, we attach.
The prefix and suffix we are saving is only 32 characters long. Which means if we have a match within a 80-char sentence, and the same sentence is repeated in an other document, we will have a false match.

> About canonical-uri: they are not in our control, we cannot force any page owner to use them.

We are also not obligated to make H work on broken pages that omit the required meta-info.

  * * *

To summarize: the query parameters typically address *different content*. We should not remove them.